test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -434,35 +469,106 @@ class test_uct_cm_sockaddr : public uct_test {
+    {
+        test_uct_cm_sockaddr *self = reinterpret_cast<test_uct_cm_sockaddr *>(arg);
+        return self->common_priv_data_cb(arg, self->m_client->max_conn_priv, priv_data);
+    }
 
-        wait_for_bits(&m_cm_state, TEST_CM_STATE_CONNECT_REQUESTED);
-        EXPECT_TRUE(m_cm_state & TEST_CM_STATE_CONNECT_REQUESTED);
+    static ssize_t server_cm_priv_data_cb(void *arg,
+                                          const uct_cm_ep_priv_data_pack_args_t
+                                          *pack_args, void *priv_data)
+    {
+        test_uct_cm_sockaddr *self = reinterpret_cast<test_uct_cm_sockaddr *>(arg);
+        return self->common_priv_data_cb(arg, self->m_server->max_conn_priv, priv_data);
+    }
+
+    void accept(uct_cm_h cm, uct_conn_request_h conn_request,
+                uct_cm_ep_server_conn_notify_callback_t notify_cb,
+                uct_ep_disconnect_cb_t disconnect_cb,
+                void *user_data)
+    {
+        uct_ep_params_t ep_params;
+        ucs_status_t status;
+        uct_ep_h ep;
+
+        ASSERT_TRUE(m_server->listener());
+        m_server->reserve_ep(m_server->num_eps());
+
+        ep_params.field_mask = UCT_EP_PARAM_FIELD_CM                     |
+                               UCT_EP_PARAM_FIELD_CONN_REQUEST           |
+                               UCT_EP_PARAM_FIELD_USER_DATA              |
+                               UCT_EP_PARAM_FIELD_SOCKADDR_CONNECT_CB    |
+                               UCT_EP_PARAM_FIELD_SOCKADDR_DISCONNECT_CB |
+                               UCT_EP_PARAM_FIELD_SOCKADDR_CB_FLAGS      |
+                               UCT_EP_PARAM_FIELD_SOCKADDR_PACK_CB;
+
+        ep_params.cm                         = cm;
+        ep_params.conn_request               = conn_request;
+        ep_params.sockaddr_cb_flags          = UCT_CB_FLAG_ASYNC;
+        ep_params.sockaddr_pack_cb           = server_cm_priv_data_cb;
+        ep_params.sockaddr_connect_cb.server = notify_cb;
+        ep_params.disconnect_cb              = disconnect_cb;
+        ep_params.user_data                  = user_data;
+
+        status = uct_ep_create(&ep_params, &ep);
+        ASSERT_UCS_OK(status);
+        m_server->eps().back().reset(ep, uct_ep_destroy);
     }
 
     virtual void server_accept(entity *server, uct_conn_request_h conn_request,
                                uct_cm_ep_server_conn_notify_callback_t notify_cb,
                                uct_ep_disconnect_cb_t disconnect_cb,
                                void *user_data)
     {
-        server->accept(server->cm(), conn_request, notify_cb, disconnect_cb,
-                       user_data);
+        accept(server->cm(), conn_request, notify_cb, disconnect_cb, user_data);
+    }
+
+    void verify_remote_data(const void *remote_data, size_t remote_length)
+    {
+        std::vector<char> r_data((char*)(remote_data), (char*)(remote_data) + remote_length);
+
+        if (remote_length == short_priv_data_len) {
```

Comment:
all tests against remote_length are redundant, if strings have different lengths they are not equal.

Reply:
but need to know to which private data to compare to the short or long one.
do you suggest to compare to both and expect that one of them is correct?

Reply:
this is why better to implement a class/func which returns priv_data by requested length...

Reply:
instead of having just two pre-initialized parameters and using them? imho it's simpler.

Reply:
for example, for TCP - it's already configurable, how to test any side?
case 2 - some transport provides priv_data length according to HW capabilities or OS setting.
IMO buffer/length should be initialized by request.

Reply:
IMO we should test the current cases if it makes things simpler now. we dont have sockaddr transports now that provide priv_data length according to HW capabilities or OS settings...

Reply:
we have TCP with not tested parameter
also tests should take into account API + documentation, two pre initialized private data buffers and lengths are based on implementation details.

Reply:
they are based on the values that the current sockaddr transports support (for tcp it's even changed in the test). it's not implementation details...

Reply:
> for tcp it's even changed in the test

then it's not pre initialized but reinitialized every time. why do we need 2 values (short and long) then?

Reply:
it's initialized in the ctor. i meant that for tcp i set a different priv_data len value in the ctor (since this value if configurable).
since tcp can support large priv_data lengths, but rdmacm cannot, i would like to test both. short for rdmacm and long for tcp_sockcm.

Reply:
in the test ctor? why do need to initialize 2 values if it's called every test and only one is used?
we still don't support 2 CMs for now, right?

Reply:
right, we dont support 2 CMs. 
even though the uct_sockaddr tests are currently testing mostly rdmacm and tcp_sockcm is skipped in most tests, i added this 'infrastructure' as preparation to when tcp_sockcm will be tested as well. i already use this code when i am testing tcp_sockcm.

Reply:
what if we will support 3 CMs? number of supported CMs is implementation details. IMO we have to generalize this.

Reply:
the third cm would be tested with either short or long. i dont think we need to prepare the test infra for a third cm that isn't in our current plans.